dataactvalidator/sqs_work_dispatcher.py

```diff
@@ -0,0 +1,756 @@
+
+            # Now that SQS message has been handled, ensure worker process is dead
+            if self._worker_process.is_alive():
+                worker = ps.Process(self._worker_process.pid)
+                # Use kill instead of multiprocess.Process.terminate() or psutil.Process.terminate(), since each of
+                # those send a signal.SIGTERM which is not as immediate and final as kill (signal.SIGKILL)
+                log_job_message(
+                    logger=self._logger,
+                    message="Message handled. Now killing worker process with PID [{}] "
+                            "as it is alive but has no more work to do.".format(worker.pid),
+                    is_debug=True
+                )
+                worker.kill()
+        finally:
+            self._handling_exit_signal = False
+            if self._dispatcher_exiting:
+                # An exit signal was received by the parent dispatcher process.
+                # Continue with exiting the parent process, as per the original signal, after having handled it
+                log_job_message(
+                    logger=self._logger,
+                    message="Exiting from parent dispatcher process with PID [{}] "
+                            "to culminate exit-handling of signal [{}]".format(os.getpid(), signal_or_human),
+                    is_debug=True
+                )
+                # Simply doing sys.exit or raise SystemExit, even with a given negative exit code won't exit this
+                # process with the negative signal value, as is the Python custom.
+                # So override the assigned handler for the signal we're handing, resetting it back to default,
+                # and kill the process using that signal to get it to exit as it would if we never handled it
+                signal.signal(signum, signal.SIG_DFL)
+                os.kill(os.getpid(), signum)
+            elif not self.allow_retries:
+                # If only the child process died, but the message it was working was moved to the Dead Letter Queue
+                # because the queue is not configured to allow retries, raise an exception so the caller
+                # might mark the job as failed with its normal error-handling process
+                raise QueueWorkerProcessError("Worker process with PID [{}] was not able to complete its job due to "
+                                              "interruption from exit signal [{}]. The queue message for that job "
+                                              "was moved to the Dead Letter Queue, where it can be reviewed for a "
+                                              "manual restart, or other remediation.".format(self._worker_process.pid,
+                                                                                             signal_or_human))
+
+    def _set_message_visibility(self, new_visibility):
+        if self._current_sqs_message is None:
```

Comment:
Can we include a basic function description here?

Reply:
Done